### PR TITLE
Generified Median.java / Performance improvements in general

### PR DESCRIPTION
**Pull request by Fuzzlemann**
Median.java
* Improves performance when comparing Integers because there is no need to convert back to a List
* Instead of Median.forInt or Median.forLong Median.forList can be used for every number
* Now every type of number can be compared
* Cleaner code in general

DecimalFormat.java
* Storing DecimalFormat rather than creating a new DecimalFormat object each time it's applied

### CODE REVIEW

File: Plan/src/main/java/com/djrapitops/plan/utilities/formatting/DecimalFormatter.java [hunk 1]
```diff
@@ -28,14 +28,14 @@
  */
 public class DecimalFormatter implements Formatter<Double> {
 
-    private final PlanConfig config;
+    private final DecimalFormat decimalFormat;
```
[the remark this answers is not in the document]
Comment (Fuzzlemann) [reply Plan/src/main/java/com/djrapitops/plan/utilities/formatting/DecimalFormatter.java]: Fix #1 applied.

File: Plan/src/test/java/com/djrapitops/plan/utilities/analysis/MedianTest.java [hunk 2]
```diff
@@ -30,23 +31,23 @@ public void simpleEven() {
...
+        double result = Median.forList(testValues).calculate();
 
         assertEquals(expected, result, 0.01);
     }
 
     @Test
     public void empty() {
         double expected = -1;
-        double result = Median.forInt(Collections.emptyList()).calculate();
+        double result = Median.forList(new ArrayList<Integer>()).calculate();
```
[the remark this answers is not in the document]
Comment (Fuzzlemann) [reply Plan/src/test/java/com/djrapitops/plan/utilities/analysis/MedianTest.java]: Collections.emptyList() does not comply with List<T extends Number & Comparable<? super T>>, this is why a specific Object is needed

File: Plan/src/main/java/com/djrapitops/plan/utilities/analysis/Median.java [hunk 3]
```diff
@@ -16,37 +16,27 @@
...
         size = values.size();
     }
 
-    public static Median forInt(Collection<Integer> integers) {
-        return new Median(integers, 0);
-    }
-
-    public static Median forLong(List<Long> longs) {
-        return new Median(longs);
+    public static <T extends Number & Comparable<? super T>> Median<T> forList(List<T> list) {
```
[the remark this answers is not in the document]
Comment (Fuzzlemann) [reply Plan/src/main/java/com/djrapitops/plan/utilities/analysis/Median.java]: Done

File: Plan/src/main/java/com/djrapitops/plan/utilities/analysis/Median.java [hunk 4]
```diff
@@ -16,37 +16,27 @@
...
 
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /**
  * Math utility for calculating the median from Integer values.
  *
  * @author Rsl1122
```
[the remark this answers is not in the document]
Comment (Fuzzlemann) [reply Plan/src/main/java/com/djrapitops/plan/utilities/analysis/Median.java]: Done

**Comment by Rsl1122** [comment]
Thanks for the changes. Changes merged to `development` and will be released under version 4.5.1 when ready.

**Pull request merged**